lib/scholar/linear/logistic_regression.ex

```diff
@@ -0,0 +1,193 @@
+defmodule Scholar.Linear.LogisticRegression do
+  @moduledoc """
+  Logistic regression in both binary and multinomial variants.
+  """
+  import Nx.Defn
+
+  @derive {Nx.Container, containers: [:coefficients, :bias], keep: [:mode]}
+  defstruct [:coefficients, :bias, :mode]
+
+  @doc """
+  Fits a logistic regression model for sample inputs `x` and sample
+  targets `y`.
+
+  Depending on number of classes the function chooses either binary
+  or multinomial logistic regression.
+  """
+  defn fit(x, y, opts \\ []) do
+    opts =
+      keyword!(opts,
+        iterations: 1000,
+        lr: 0.01,
+        num_classes: 1
+      )
```

Comment:
We should document the options.

Reply:
Is this documentation sufficient @jonatanklosko?